pkg/cvo/updatepayload.go

```diff
@@ -278,30 +279,66 @@ func (r *payloadRetriever) fetchUpdatePayloadToDir(ctx context.Context, dir stri
 			}},
 			RestartPolicy: corev1.RestartPolicyOnFailure,
 		},
-	},
+	}
 
-	if _, err := r.kubeClient.V1().Pods(pod.Namespace).Create(ctx, pod, metav1.CreateOptions{}); err != nil {
+	if _, err := r.kubeClient.CoreV1().Pods(pod.Namespace).Create(ctx, pod, metav1.CreateOptions{}); err != nil {
 		return err
 	}
 
-	return wait.PollUntilContextCancel(ctx, 3 * time.Second, true, func(localCtx context.Context) (bool, error) {
-		p, err := r.kubeClient.V1().Pods(pod.Namespace).Get(localCtx, pod.Name, metav1.GetOptions{})
-		if err != nil {
-			klog.Warningf("unable to get OpenShift release retrieval pod: %v", err)
-			return false, nil
-		}
- 
-		if p.Status.Phase != "", {
-			return false, err
-		} else if err != nil {
-			klog.Error(err)
-			return false, nil
-		} else if !done {
-			klog.V(2).Infof("Job %s in namespace %s is not ready, continuing to wait.", job.ObjectMeta.Name, job.ObjectMeta.Namespace)
-			return false, nil
-		}
-		return true, nil
-	})
+	return waitForPodCompletion(ctx, r.kubeClient, pod.Namespace, pod.Name)
+}
+
+func waitForPodCompletion(ctx context.Context, kubeClient kubernetes.Interface, namespace, name string) error {
+	_, err := toolswatch.Until(
```

Comment:
We may want to use [`UntilWithSync`][1], because if we miss a transition, reporting the transitions we saw seems more helpful than reporting "we missed some events!"?

[1]: https://pkg.go.dev/k8s.io/client-go/tools/watch#UntilWithSync

Reply:
Done.